src/parser/smt2/smt2.cpp

```diff
@@ -1235,16 +1233,45 @@ Expr Smt2::makeSygusBoundVarList(Datatype& dt,
   return getExprManager()->mkExpr(kind::BOUND_VAR_LIST, lvars);
 }
 
-const void Smt2::getSygusPrimedVars( std::vector<Expr>& vars, bool isPrimed ) {
-  for (unsigned i = 0, size = d_sygusInvVars.size(); i < size; i++)
+const void Smt2::getSygusInvVars(FunctionType t,
+                                 std::vector<Expr>& vars,
+                                 std::vector<Expr>& primed_vars)
+{
+  std::vector<Type> argTypes = t.getArgTypes();
+  std::map<Type, unsigned> type_count;
+  for (const Type& ti : argTypes)
   {
-    Expr v = d_sygusInvVars[i];
-    std::map< Expr, bool >::iterator it = d_sygusVarPrimed.find( v );
-    if( it!=d_sygusVarPrimed.end() ){
-      if( it->second==isPrimed ){
-        vars.push_back( v );
+    if (type_count.find(ti) == type_count.end())
+    {
+      type_count[ti] = 0;
+    }
+    std::stringstream ss;
+    ss << ti << "_" << type_count[ti]++;
```

Comment:
This is a pretty nasty name for this variable. I would leave these bound variables have the default names actually.

Reply:
Also more generally, notice that name clashes do not need to be resolved by the caller of mkBoundVar, mkSkolem, since we automatically append _N for some internal N, so the type_count is not strictly necessary in these kind of cases.